src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +316,1319 @@ const ActorInit En_Egol_InitVars = {
+        quakeSize = 4 - (s32)(fabsf(player->actor.world.pos.y - this->actor.world.pos.y) * 0.02f);
+        if (quakeSize > 4) {
+            quakeSize = 4;
+        } else if (quakeSize < 1) {
+            quakeSize = 1;
+        }
+
+        if (player->stateFlags3 != PLAYER_STATE3_1000000) {
+            func_800BC848(&this->actor, play, quakeSize, 2);
+        }
+        if (Animation_OnFrame(&this->skelAnime, leftFootFrame)) {
+            Math_Vec3f_Copy(&spawnPos, &this->leftFootPos);
+            spawnPos.y = this->actor.floorHeight;
+            Actor_SpawnFloorDustRing(play, &this->actor, &spawnPos, 0.0f, 10, 6.0f, 50, 30, true);
+        } else {
+            Math_Vec3f_Copy(&spawnPos, &this->rightFootPos);
+            spawnPos.y = this->actor.floorHeight;
+            Actor_SpawnFloorDustRing(play, &this->actor, &spawnPos, 0.0f, 10, 6.0f, 50, 30, true);
+        }
+    }
+}
+
+void EnEgol_DestroyBlocks(EnEgol* this, PlayState* play, Vec3f pos1, Vec3f pos2) {
+    Actor* egBlock = play->actorCtx.actorLists[ACTORCAT_PROP].first;
+
+    while (egBlock != NULL) {
+        if (egBlock->id != ACTOR_EN_EGBLOCK) {
+            egBlock = egBlock->next;
+            continue;
+        }
+        if (egBlock->colChkInfo.health > 0) {
+            Vec3f blockTo1;
+            Vec3f blockTo2;
+            f32 dist1;
+            f32 dist2;
+
+            blockTo1.x = pos1.x - egBlock->world.pos.x;
+            blockTo1.y = pos1.y - egBlock->world.pos.y;
+            blockTo1.z = pos1.z - egBlock->world.pos.z;
+            blockTo2.x = pos2.x - egBlock->world.pos.x;
+            blockTo2.y = pos2.y - egBlock->world.pos.y;
+            blockTo2.z = pos2.z - egBlock->world.pos.z;
+
+            dist1 = sqrtf(SQXZ(blockTo1));
+            dist2 = sqrtf(SQXZ(blockTo2));
+
+            if ((dist1 < 40.0f) || (dist2 < 40.0f)) {
+                dist1 = sqrtf(SQ(blockTo1.y));
+                dist2 = sqrtf(SQ(blockTo2.y));
+                if ((dist1 < 40.0f) || (dist2 < 40.0f)) {
+                    egBlock->colChkInfo.health = 0;
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_WALL_BROKEN);
+                    break;
+                }
+            }
+        }
+        egBlock = egBlock->next;
+    }
+}
+
+void EnEgol_GetWaypoint(EnEgol* this) {
+    if ((this->pathIndex != -1) && (this->path != NULL) &&
+        !SubS_CopyPointFromPath(this->path, this->waypoint, &this->waypointPos)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+#define EYEGORE_SET_DIM(element, centerx, centery, centerz, radiusv, scalev) \
+    element.dim.modelSphere.radius = radiusv;                                \
+    element.dim.scale = scalev;                                              \
+    element.dim.modelSphere.center.x = centerx;                              \
+    element.dim.modelSphere.center.y = centery;                              \
+    element.dim.modelSphere.center.z = centerz
```

Comment:
The arguments should still have camelCase names even in a macro. It'd also be better to be specific that it's sphere dimensions.